### PR TITLE
Only show auth0 part of the profile in the output for auth0 output

### PR DESCRIPTION
**Pull request by gdestuynder**
Change group handling for backward compat and clarity on group quality.

The idea being this is:

"groups" are self managed groups in either LDAP or Mozillians.org and
let RPs decide how to best manage access themselves. For ex. discourse
can create a Mozillians.org group "IAM" and both manage and use it to
restrict access to certain discussions to accounts in that group.

"authoritative_groups" are strongly managed groups that are emitted by
IAM directly. This includes the "moco", "mofo" groups (ie Staff members)
that is sourced from workday/servicenow, the "nda" group which is well
managed and sourced from mozillians.org  and all the auth0 service
groups which are created for each RP.

Authentication is done via 2 layers:
- Auth0: allows user to login if the user is in any of the authoritative
groups - if none is required, just allow login. When logging the user
in, the "lastUsed" attribute is also updated to reflect the last time
the user used that group.
If "lastUsed" is older than 3month, access is denied and the user must
re-request access

- RP: the rp can use any of the "groups" or "authoritative_groups" at
it's entire discretion for additional access management purposes.

### CODE REVIEW

File: doc/User_profile.json [hunk 1]
```diff
@@ -79,56 +79,35 @@
 	],
 	"picture": "https://s.gravatar.com/avatar/...png",
 	"shirtSize": "L",
-	"membership": {
-		"Workday": [
+	"groups": [
```
Comment (johngian) [inline doc/User_profile.json:82]: Is there a reason behind the difference in the format of ``groups`` and ``authoritativeGroups``? I suggest we keep it consistent across all groups:

``
"groups": [
  {
    "name": "foobar",
    "created":  "",
    "lastUsed": "",
    ...
  },
  ..  
]
``

**Review by johngian — CHANGES_REQUESTED** [review]

File: doc/User_profile.json (hunk 1 above)
Comment (gdestuynder) [reply doc/User_profile.json:82]: yes, authoritativegroups need extra attributes that regular groups dont care for
the authoritative groups would be created by auth0 or directly by CIS (ie adding a new authoritative group that is mozillians-managed is manual)

Comment (gdestuynder) [reply doc/User_profile.json:82]: note @johngian due to this i believe no change to the PR is needed
this schema was discussed a couple of weeks back (with you :) and is also explained above

I can also add this in the readme though.  https://github.com/mozilla-iam/cis/issues/48 references all this as well

Comment (johngian) [reply doc/User_profile.json:82]: sound good

**Review by johngian — APPROVED** [review]

**Review by akatsoulas — APPROVED** [review]

**Pull request merged**